server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -231,4 +243,24 @@ private String convertToString(ArrayList<String> comments) {
 
     return res.toString();
   }
+
+  /**
+   * Filters out a comment string so that it's readable and doesn't break the frontend.
+   *
+   * @param comment The comment to be filtered.
+   * @return A properly formatted comment
+   */
+  private String filterComment(String comment) {
+    String filteredComment = comment;
+
+    // Handles non-ASCII characters, newlines,
+    // trimmed quotes, embedded escaped quotes.
+    // Order matters.
+    filteredComment = filteredComment.replaceAll("[^\\x00-\\x7F]", "");
+    filteredComment = String.join("\n", filteredComment.split("\\\\n"));
+    filteredComment = filteredComment.substring(1, filteredComment.length() - 1);
+    filteredComment = filteredComment.replace("\\\"", "");
```

Comment:
Is the analysis supposed to work on non-English text? Because filtering out all non-ASCII characters will create a lot of problems with that.

Reply:
Good call. Will take out non-ASCII and replace with just emojis